engine/src/test/java/io/camunda/zeebe/engine/util/StreamProcessingComposite.java

```diff
@@ -136,121 +146,135 @@ public RecordStream events() {
-        .key(key)
-        .recordType(RecordType.COMMAND)
-        .intent(intent)
-        .event(value)
-        .write();
+    final var writer =
+        streams
+            .newRecord(getLogName(partition))
+            .key(key)
+            .recordType(RecordType.COMMAND)
+            .intent(intent)
+            .event(value);
+    return writeActor.submit(writer::write).join();
   }
 
   public long writeCommand(final long key, final Intent intent, final UnpackedObject value) {
-    return streams
-        .newRecord(getLogName(partitionId))
-        .recordType(RecordType.COMMAND)
-        .key(key)
-        .intent(intent)
-        .event(value)
-        .write();
+    final var writer =
+        streams
+            .newRecord(getLogName(partitionId))
+            .recordType(RecordType.COMMAND)
+            .key(key)
+            .intent(intent)
+            .event(value);
+    return writeActor.submit(writer::write).join();
   }
 
   public long writeCommand(final Intent intent, final UnpackedObject value) {
-    return streams
-        .newRecord(getLogName(partitionId))
-        .recordType(RecordType.COMMAND)
-        .intent(intent)
-        .event(value)
-        .write();
+    final var writer =
+        streams
+            .newRecord(getLogName(partitionId))
+            .recordType(RecordType.COMMAND)
+            .intent(intent)
+            .event(value);
+    return writeActor.submit(writer::write).join();
   }
 
   public long writeCommand(
       final int requestStreamId,
       final long requestId,
       final Intent intent,
       final UnpackedObject value) {
-    return streams
-        .newRecord(getLogName(partitionId))
-        .recordType(RecordType.COMMAND)
-        .requestId(requestId)
-        .requestStreamId(requestStreamId)
-        .intent(intent)
-        .event(value)
-        .write();
+    final var writer =
+        streams
+            .newRecord(getLogName(partitionId))
+            .recordType(RecordType.COMMAND)
+            .requestId(requestId)
+            .requestStreamId(requestStreamId)
+            .intent(intent)
+            .event(value);
+    return writeActor.submit(writer::write).join();
   }
 
   public long writeCommandRejection(final Intent intent, final UnpackedObject value) {
-    return streams
-        .newRecord(getLogName(partitionId))
-        .recordType(RecordType.COMMAND_REJECTION)
-        .intent(intent)
-        .event(value)
-        .write();
+    final var writer =
+        streams
+            .newRecord(getLogName(partitionId))
+            .recordType(RecordType.COMMAND_REJECTION)
+            .intent(intent)
+            .event(value);
+    return writeActor.submit(writer::write).join();
   }
 
   public static String getLogName(final int partitionId) {
     return STREAM_NAME + partitionId;
   }
 
+  /** Used to run writes within an actor thread. */
+  private static final class WriteOnActorHelper extends Actor {
```

Comment:
🔧  I was thinking about this name a bit, I think it actually is an Actor, not a helper. So I suggest changing the name to `WriteActor` or `WriterActor` or `WritingActor`, etc.